yang/vyatta-protocols-frr-link-params-v1.yang

```diff
@@ -0,0 +1,194 @@
+module vyatta-protocols-frr-link-params-v1 {
+	namespace "urn:vyatta.com:mgmt:vyatta-protocols-frr-link-params:1";
+	prefix vyatta-protocols-frr-link-params-v1;
+
+	import vyatta-types-v1 {
+		prefix types;
+	}
+	import configd-v1 {
+		prefix configd;
+	}
+	import vyatta-interfaces-v1 {
+		prefix if;
+	}
+	import vyatta-interfaces-dataplane-v1 {
+		prefix interfaces-dataplane;
+	}
+	import vyatta-interfaces-loopback-v1 {
+		prefix interfaces-loopback;
+	}
+	import vyatta-interfaces-switch-v1 {
+		prefix interfaces-switch;
+	}
+
+	organization "Niral Networks Private Limited";
+	contact
+		"Niral Networks
+		Postal: Sigma Soft Tech Park, Whitefield Main Road,
+		Bengaluru, India 560066
+		Web: www.niralnetworks.com";
+
+	description
+		"Copyright (c) 2019-2020, Niral Networks Intellectual Property.
+		All rights reserved.
+
+		Redistribution and use in source and binary forms, with or
+		without modification, are permitted provided that the following
+		conditions are met:
+
+		1. Redistributions of source code must retain the above copyright
+		notice, this list of conditions and the following disclaimer.
+		2. Redistributions in binary form must reproduce the above
+		copyright notice, this list of conditions and the following
+		disclaimer in the documentation and/or other materials provided
+		with the distribution.
+		3. Neither the name of the copyright holder nor the names of its
+		contributors may be used to endorse or promote products derived
+		from this software without specific prior written permission.
+
+		THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+		'AS IS' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+		LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+		FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+		COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+		INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+		BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+		LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+		CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+		LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+		ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+		POSSIBILITY OF SUCH DAMAGE.
+
+		SPDX-License-Identifier: BSD-3-Clause
+		The YANG module package for vyatta-protocols-frr-link-params-v1";
+
+	revision 2021-03-02 {
+		description 
+			"Added link-params option.";
+	}
+	
+	grouping link-params {
+		container link-params {
+			configd:help "Configure interface link parameters";
+			leaf admin-group {
+				type string {
+					pattern "0[Xx][a-fA-F0-9]+";
+				}
+				must "(../enable)" {
+					error-message "link-params enable must be set.";
+				}
+				configd:help "Administrative group membership";
+				description "Administrative group membership";
+			}
+			leaf available-bandwidth {
+				type uint32;
```

Comment:
Much cleaner. Please add units statements to all of these so it's clear what units are being used. Maybe also mention the units in the configd:help as units aren't exposed via the CLI help.

Reply:
uint64 would be able to allocate more bandwidth than uint32.. It has been modified

Reply:
Please push changes - presumably you will be modifying ALL bandwidth nodes to uint64?

Reply:
I've pushed my changes. In all the bandwidth I've modified to uint64